src/otaclient/ota_core.py

```diff
@@ -1152,21 +1162,16 @@ def update(self, request: UpdateRequestV2) -> None:
                 failure_reason=e.get_failure_reason(),
                 failure_type=e.failure_type,
             )
+            self._exit_from_dynamic_client()
 
     def client_update(self, request: ClientUpdateRequestV2) -> None:
         """
         NOTE that client update API will not raise any exceptions. The failure information
             is available via status API.
         """
         if _env.is_dynamic_client_running():
-            self._live_ota_status = OTAStatus.FAILURE
-            e = ota_errors.DuplicatedClientUpdateRequest(module=__name__)
-            self._on_failure(
-                e,
-                ota_status=OTAStatus.FAILURE,
-                failure_reason=e.get_failure_reason(),
-                failure_type=e.failure_type,
-            )
+            # Duplicates client update should not be allowed.
+            # TODO(airkei) [2025-06-19]: should return the dedicated error code for "client update"
```

Comment:
Not set status in client update.
When duplicated `ClientUpdate` is called, just return without do anything.